src/ucs/datastruct/linear_func.h

```diff
@@ -57,14 +57,100 @@ ucs_linear_func_apply(ucs_linear_func_t f, double x)
 /**
  * Sum two linear functions.
  *
- * @param [out] result   Filled with the resulting linear function.
  * @param [in]  func1    First function to add.
  * @param [in]  func2    Second function to add.
+ *
+ * @return Linear function representing (func1 + func2)
  */
 static UCS_F_ALWAYS_INLINE ucs_linear_func_t
 ucs_linear_func_add(ucs_linear_func_t func1, ucs_linear_func_t func2)
 {
     return ucs_linear_func_make(func1.c + func2.c, func1.m + func2.m);
 }
 
+
+/**
+ * Subtract one linear function from another.
+ *
+ * @param [in]  func1    Linear function to subtract from.
+ * @param [in]  func2    Linear function to subtract.
+ *
+ * @return Linear function representing (func1 - func2)
+ */
+static inline ucs_linear_func_t
+ucs_linear_func_sub(ucs_linear_func_t func1, ucs_linear_func_t func2)
+{
+    return ucs_linear_func_make(func1.c - func2.c, func1.m - func2.m);
+}
+
+
+/**
+ * Add one linear function to another in-place.
+ *
+ * @param [inout]  func1    First linear function to add, and the result of the
+ *                          operation
+ * @param [in]     func2    Second function to add.
+ */
+static inline void
+ucs_linear_func_add_inplace(ucs_linear_func_t *func1, ucs_linear_func_t func2)
```

Comment:
why not use `f1 = ucs_linear_func_add(f1, f2)`?

Reply:
it's possible as well, but in some usage cases it's shorter to write as something like f1 += f2